python/wikibot/wikibot.py

```diff
@@ -100,14 +127,43 @@ def __init__(self):
                                    day=today.day)
     self.time_claim.setTarget(time_target)
 
+    self.unique_properties = {self.date_of_birth, self.date_of_death}
+    kb = self.store
+    # Collect unique-valued properties.
+    # They will be used to compute CONFLICT counts.
+    constraint_role = kb["P2302"]
+    unique = kb["Q19474404"]         # single-value constraint
+    #w_time = kb["/w/time"]
+    #w_item = kb["/w/item"]
+    p_subproperty_of = kb["P1647"]
+    p_location = kb["P276"]
+    for prop in kb["/w/entity"]("role"):
+      for constraint_type in prop(constraint_role):
+        if constraint_type == unique or constraint_type["is"] == unique:
+          self.unique_properties.add(prop)
+    self.unique_cache = {}
+
+  def is_unique(self, prop):
+    if prop not in self.unique_cache:
+      self.unique_cache[prop] = (prop in self.unique_properties)
+    return self.unique_cache[prop]
```

Comment:
Both unique_cache and unique_properties have the same complexity for lookup, so there is no added benefit in keeping unique_cache. Maybe just simplify is_unique to:
```
def is_unique(self, prop):
   return prop in self.unique_properties
```